backtesting/backtesting.py

```diff
@@ -1514,3 +1525,68 @@ def plot(self, *, results: pd.Series = None, filename=None, plot_width=None,
+                       maximize: Union[str, Callable[[pd.Series], float]] = 'SQN',
+                       max_tries: Union[int, float] = 100,
+                       **kwargs) -> Union[pd.Series, Tuple[pd.Series, pd.Series]]:
+        """
+        Optimize strategy parameters to an optimal combination using
+        scikit-optimize. Returns result `pd.Series` of
+        the best run.
+
+        `maximize` is a string key from the
+        `backtesting.backtesting.Backtest.run`-returned results series,
+        or a function that accepts this series object and returns a number;
+        the higher the better. By default, the method maximizes
+        Van Tharp's [System Quality Number](https://google.com/search?q=System+Quality+Number).
+
+        Additional keyword arguments represent strategy arguments with
+        list-like collections of possible values. For example, the following
+        code finds and returns the "best" of the 7 admissible (of the
+        9 possible) parameter combinations:
+
+            backtest.optimize(sma1=[5, 10, 15], sma2=[10, 20, 40],
+                              constraint=lambda p: p.sma1 < p.sma2)
+
+        .. TODO::
+            Improve multiprocessing with joblib.parallel .
+        """
+
+        if not kwargs:
+            raise ValueError('Need some strategy parameters to optimize')
+
+        maximize_key = None
+        if isinstance(maximize, str):
+            maximize_key = str(maximize)
+            stats = self._results if self._results is not None else self.run()
+            if maximize not in stats:
+                raise ValueError('`maximize`, if str, must match a key in pd.Series '
+                                 'result of backtest.run()')
+
+            def maximize(stats: pd.Series, _key=maximize):
+                return stats[_key]
+
+        elif not callable(maximize):
+            raise TypeError('`maximize` must be str (a field of backtest.run() result '
+                            'Series) or a function that accepts result Series '
+                            'and returns a number; the higher the better')
+
+        dimensions = [Integer(name='n1', low=min(kwargs.get('n1')), high=max(kwargs.get('n1'))),
+                      Integer(name='n2', low=min(kwargs.get('n2')), high=max(kwargs.get('n2')))]
+
+        @use_named_args(dimensions=dimensions)
+        def convert(n1, n2):
+            res = self.run(**dict(zip(['n1', 'n2'], [n1, n2])))
+            return -res[maximize_key]
+
+        res = forest_minimize(func=convert,
+                              dimensions=dimensions,
+                              n_calls=max_tries, base_estimator="ET",
+                              random_state=4)
+
+        self.run(**dict(zip(['n1', 'n2'], res.x)))
+        return_result = pd.Series([res.fun, res.x, res, self._results], index=[
+                                  "best_fitness", "best_parameters", "full_results", "stats_best"])
+
+        return return_result
```

Comment:
Return from `Backtest.optimize()` should remain best-run `stats: pd.Series`, but you are encouraged to add another method argument:
```py
return_optimization: bool = False
```
(akin to `return_heatmap=`) which additionally returns `scipy.optimize.OptimizeResult` as obtained from `forest_minimize()`. The user can then use `skopt.plots.plot_objective()` or the object's defined structure to draw further insights.